connector/kafka-0-10/src/test/scala/org/apache/spark/streaming/kafka010/KafkaRDDSuite.scala

```diff
@@ -117,7 +118,7 @@ class KafkaRDDSuite extends SparkFunSuite {
     )
     messages.foreach { case (k, v) =>
       val record = new SimpleRecord(k.getBytes, v.getBytes)
-      log.appendAsLeader(MemoryRecords.withRecords(CompressionType.NONE, record), 0);
+      log.appendAsLeader(MemoryRecords.withRecords(Compression.NONE, Seq(record): _*), 0);
     }
```

Comment:
In v3.8.0, the signature of method `withRecords` has been changed to `MemoryRecords withRecords(Compression compression, ...)`

- v3.7.1
https://github.com/apache/kafka/blob/3.7.1/clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java#L594
<img width="729" alt="image" src="https://github.com/user-attachments/assets/1990a695-336b-4ac5-8ae6-c5790b84c687">

- v3.8.0
https://github.com/apache/kafka/blob/3.8.0/clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java#L597
<img width="693" alt="image" src="https://github.com/user-attachments/assets/3a55001d-b184-41ec-84be-777e30575bb3">